drivers/pulseaudio/audio_driver_pulseaudio.cpp

```diff
@@ -156,38 +156,45 @@ Error AudioDriverPulseAudio::init() {
 	exit_thread = false;
 
 	mix_rate = GLOBAL_DEF("audio/mix_rate", DEFAULT_MIX_RATE);
-	channels = detect_channels();
+	pa_channels = detect_channels();
+
+	switch (pa_channels) {
+		case 1: // Mono
+		case 3: // Surround 2.1
+		case 5: // Surround 5.0
+		case 7: // Surround 7.0
+			channels = pa_channels + 1;
+			break;
 
-	switch (channels) {
 		case 2: // Stereo
-		case 4: // Surround 3.1
+		case 4: // Surround 4.0
 		case 6: // Surround 5.1
 		case 8: // Surround 7.1
+			channels = pa_channels;
 			break;
 
 		default:
-			ERR_PRINTS("PulseAudio: Unsupported number of channels: " + itos(channels));
+			ERR_PRINTS("PulseAudio: Unsupported number of channels: " + itos(pa_channels));
 			ERR_FAIL_V(ERR_CANT_OPEN);
 			break;
 	}
 
 	pa_sample_spec spec;
 	spec.format = PA_SAMPLE_S16LE;
-	spec.channels = channels;
+	spec.channels = pa_channels;
 	spec.rate = mix_rate;
 
 	int latency = GLOBAL_DEF("audio/output_latency", DEFAULT_OUTPUT_LATENCY);
 	buffer_frames = closest_power_of_2(latency * mix_rate / 1000);
-	buffer_size = buffer_frames * channels;
```

Comment:
It might make sense to keep storing `buffer_size` and a new `pa_buffer_size` to avoid recomputing it all the time in `thread_func`?

Maybe also have an explanatory comment about the difference between `pa_channels` and `channels` as it's super clear from reading (apart from the fact that one is used for output and the other one for input).

Reply:
Sounds good to me, I'll update the PR with these changes.